SeriyBg/fn-training-project
# Implemented Tasks core, sshclien, basics models

Was tested on local machine only with valid ssh and valid credentials.
Unit tests is not provided.
DB is not implemented.
Logging system System.out.printf only.
Any additional input validation is not implemented.
Exceptions handling is not implemented.

Was parsed only IP Interfaces and saved to local ArrayList. 

Test scenario:
1. Run application on local machine
2. Send POST request.
    Example:
        localhost:8086/host/get_info/
        {
	    "addr": "127.0.0.1",
	    "user":"okolka",
	    "password":"jnjnuh"
        }
    Expected response:
      {
          "id": 1,
          "status": "INPROGRESS"
      }

Console output:
SshClient:<constructor> host: 127.0.0.1; user: okolka; password: jnjnuh
TaskExecutor: got new Task
TaskExecutor: Pool size: 1
TaskExecutor: Active count: 1
TaskExecutor: Task count: 1
TaskExecutor: Completed tasks: 1
ssh connected 'okolka' by '127.0.0.1'
enp0s31f6 Link encap:Ethernet  HWaddr fc:3f:db:87:7b:27  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:5330773 errors:1087 dropped:1 overruns:0 frame:547
          TX packets:2804056 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:6887636250 (6.8 GB)  TX bytes:282094082 (282.0 MB)
          Interrupt:16 Memory:f1200000-f1220000 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:418828 errors:0 dropped:0 overruns:0 frame:0
          TX packets:418828 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:68653553 (68.6 MB)  TX bytes:68653553 (68.6 MB)

wlp1s0    Link encap:Ethernet  HWaddr 44:85:00:20:2e:0c  
          inet addr:192.168.1.101  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: fe80::ffef:f438:e178:d123/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:17117529 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8065278 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:23842066696 (23.8 GB)  TX bytes:983173995 (983.1 MB)

[IpInterface(name=enp0s31f6, macAddress=fc:3f:db:87:7b:27, ipAddress=null, MTU=1500), IpInterface(name=wlp1s0, macAddress=44:85:00:20:2e:0c, ipAddress=IpAddress(addr=192.168.1.101, netmask=255.255.255.0), MTU=1500)]
RUN task  parentid: 1; id: 1
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         192.168.1.100   0.0.0.0         UG    600    0        0 wlp1s0
link-local      *               255.255.0.0     U     1000   0        0 wlp1s0
192.168.1.0     *               255.255.255.0   U     600    0        0 wlp1s0
RUN task  parentid: 1; id: 2
